File: Application/src/main/java/com/shareyourproxy/app/UserContactActivity.java
package com.shareyourproxy.app;

import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.view.MenuItem;

import com.shareyourproxy.IntentLauncher;
import com.shareyourproxy.api.domain.model.User;
import com.shareyourproxy.app.fragment.AggregateFeedFragment;
import com.shareyourproxy.app.fragment.ContactProfileFragment;

import timber.log.Timber;

import static com.shareyourproxy.Constants.ARG_USER_SELECTED_PROFILE;

/**
 * Activity that handles displaying a {@link User} profile.
 */
public class UserContactActivity extends BaseActivity {

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        ActivityCompat.finishAfterTransition(this);
        //if we launched from a notification go back to the AggregateFeedActivity explicitly
        if (this.isTaskRoot()) {
            IntentLauncher.launchMainActivity(this, AggregateFeedFragment.ARG_SELECT_CONTACTS_TAB, false, null);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        preventStatusBarFlash(this);
        if (savedInstanceState == null) {
            User user = getUserExtra();
            getSupportFragmentManager().beginTransaction()
                .replace(android.R.id.content,
                    ContactProfileFragment.newInstance(user, getLoggedInUser().id)).commit();
        }
    }

    /**
     * Get parceled user.
     */
    private User getUserExtra() {
        return getIntent().getExtras().getParcelable(ARG_USER_SELECTED_PROFILE);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
            default:
                Timber.e("Option item selected is unknown");
        }
        return super.onOptionsItemSelected(item);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/app/UserContactActivity.kt
package com.shareyourproxy.app

import android.os.Bundle
import android.support.v4.app.ActivityCompat
import android.view.MenuItem
import com.shareyourproxy.Constants.ARG_USER_SELECTED_PROFILE
import com.shareyourproxy.IntentLauncher.launchMainActivity
import com.shareyourproxy.api.domain.model.User
import com.shareyourproxy.app.fragment.AggregateFeedFragment
import com.shareyourproxy.app.fragment.ContactProfileFragment
import com.shareyourproxy.util.ButterKnife.LazyVal
import timber.log.Timber

/**
 * Activity that handles displaying a [User] profile.
 */
internal final class UserContactActivity : BaseActivity() {
    private val userExtra: User by LazyVal { intent.extras.getParcelable<User>(ARG_USER_SELECTED_PROFILE) }

    override fun onBackPressed() {
        super.onBackPressed()
        ActivityCompat.finishAfterTransition(this)
        //if we launched from a notification go back to the AggregateFeedActivity explicitly
        if (this.isTaskRoot) {
            launchMainActivity(this, AggregateFeedFragment.ARG_SELECT_CONTACTS_TAB, false, null)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        preventStatusBarFlash(this)
        if (savedInstanceState == null) {
            val user = userExtra
            supportFragmentManager.beginTransaction().replace(android.R.id.content, ContactProfileFragment(user, loggedInUser.id)).commit()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            else -> Timber.e("Option item selected is unknown")
        }
        return super.onOptionsItemSelected(item)
    }
}
